frontend/packages/erd-core/src/features/erd/components/ERDRenderer/LeftPane/LeftPane.tsx

```diff
@@ -133,29 +116,35 @@ export const LeftPane = () => {
           <SidebarGroupLabel className={styles.groupLabel} asChild>
             <span>Tables</span>
             <span className={styles.tablesHeaderRow}>
-              <button
-                type="button"
-                className={styles.showAllButton}
-                aria-label={
-                  visibleCount === allCount
-                    ? 'Hide All Tables'
-                    : 'Show All Tables'
-                }
-                tabIndex={0}
-                onClick={showOrHideAllNodes}
-              >
-                {visibleCount === allCount ? (
+              {visibilityStatus === 'all-visible' ? (
+                <button
+                  type="button"
+                  className={styles.showAllButton}
+                  aria-label="Hide All Tables"
+                  tabIndex={0}
+                  onClick={hideAllNodes}
+                >
                   <EyeOff className={styles.icon} />
-                ) : (
+                  <span className={styles.showAllText}>Hide All</span>
+                  <span className={styles.visibleCount}>
+                    ({visibleCount}/{allCount} visible)
+                  </span>
+                </button>
+              ) : (
+                <button
+                  type="button"
+                  className={styles.showAllButton}
+                  aria-label="Show All Tables"
+                  tabIndex={0}
+                  onClick={showAllNodes}
+                >
                   <Eye className={styles.icon} />
-                )}
-                <span className={styles.showAllText}>
-                  {visibleCount === allCount ? 'Hide All' : 'Show All'}
-                </span>
-                <span className={styles.visibleCount}>
-                  ({visibleCount}/{allCount} visible)
-                </span>
-              </button>
```

Comment:
I noticed there were many `visibleCount === allCount` checks, so I moved that logic to a single top-level condition and split the button into separate “Show All” and “Hide All” buttons. I also separated the `showOrHideAllNodes` function into dedicated show and hide functions, which simplified the overall logic.